addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -98,23 +110,65 @@ public static boolean contains(String tag) {
                 .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
                 .forEach(
-                        element ->
-                                newAlert()
-                                        .setRisk(Alert.RISK_MEDIUM)
-                                        .setConfidence(Alert.CONFIDENCE_HIGH)
-                                        .setDescription(getString("desc"))
-                                        .setSolution(getString("soln"))
-                                        .setReference(getString("refs"))
-                                        .setEvidence(element.toString())
-                                        .setCweId(345) // CWE-345: Insufficient Verification of Data
-                                        // Authenticity
-                                        .setWascId(15) // Application Misconfiguration
-                                        .raise());
+                        element -> {
+                            String integrityHash = calculateIntegrityHash(msg, element, tree);
+                            String otherInfo =
+                                    integrityHash.isEmpty()
+                                            ? ""
+                                            : Constant.messages.getString(
+                                                    MESSAGE_PREFIX + "otherinfo", integrityHash);
+                            newAlert()
+                                    .setRisk(Alert.RISK_MEDIUM)
+                                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                                    .setDescription(getString("desc"))
+                                    .setSolution(getString("soln"))
+                                    .setReference(getString("refs"))
+                                    .setEvidence(element.toString())
+                                    .setCweId(345) // CWE-345: Insufficient Verification of Data
+                                    // Authenticity
+                                    .setWascId(15) // Application Misconfiguration
+                                    .setOtherInfo(otherInfo)
+                                    .raise();
+                        });
+    }
+
+    private String calculateIntegrityHash(HttpMessage msg, Element element, SiteMap tree) {
+        String src = element.getAttributeValue("src");
+        String integrityHash = "";
+        if (src != null) {
+            URI newURI = null;
+
+            try {
+                String newuri = msg.getRequestHeader().getURI().toString();
+                newURI = new URI(newuri);
+                newURI = newURI.resolve(src);
+                SiteNode node =
+                        tree.findNode(
+                                new org.apache.commons.httpclient.URI(newURI.toString(), true));
+                if (node != null) {
+                    integrityHash =
```

Comment:
If this was really meant to just provide the digest (hexed) the other info should be updated to clarify that (and include the algorithm used, since there are several allowed).

Reply:
> This (the digest) should be Base64 encoded, also, better to include the algorithm to match an actual integrity value.

I did suggest this in the very beginning though :)

Reply:
> If this was really meant to just provide the digest (hexed) the other info should be updated to clarify that (and include the algorithm used, since there are several allowed).

Yes I felt this should be included, but I just waited for your ( or any other reviewers ) opinion for it. Also this means there will be UnitTests changes :). I will make the changes accordingly.

Reply:
I think I missed that exchange.

Reply:
This one is my fault too. I didn’t put the base64 detail together with the spec, sorry

Reply:
The algorithm is not being added yet.